keras_hub/src/models/segformer/segformer_tests.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2024 The KerasHub Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
```

Comment:
add separate test files for the backbone and the task.
Please add predefined backbone test routine `self.run_vision_backbone_test` - example : https://github.com/keras-team/keras-hub/blob/master/keras_hub/src/models/resnet/resnet_backbone_test.py#L30
and a predefine task test routine `self.run_task_test` and `self.run_model_saving_test`

Reply:
Added, though the `image_encoder` is causing some issues through deserialization.

It looks like it gets serialized as expected while saving, but when recovering the parent model from the config, the nested `image_encoder` config doesn't seem to get deserialized back into a `keras.Model` and stays a config dictionary.

It instantiates the SegFormer arch, but doesn't deserialize `image_encoder`:

```
ValueError: Argument `image_encoder` must be a `keras.layers.Layer` instance  or `keras.Model`. Received instead image_encoder={'module': 'keras_hub.src.models.mix_transformer.mix_transformer_backbone', 'class_name': 'MiTBackbone', 'config': {'name': 'mi_t_backbone_3', 'trainable': True, 'depths': [2, 2], 'hidden_dims': [32, 64], 'image_shape': (224, 224, 3), 'num_layers': 2, 'blockwise_num_heads': [1, 2], 'blockwise_sr_ratios': [8, 4], 'max_drop_path_rate': 0.1, 'patch_sizes': [7, 3], 'strides': [4, 2]}, 'registered_name': 'keras_hub>MiTBackbone', 'build_config': {'input_shape': None}} (of type <class 'dict'>).
```

Reply:
Found the issue - overriding `from_config()` in which we explicitly deserialize the image encoder correctly deserializes it when reviving from a config